subsys/bluetooth/mesh/cfg_cli.c

```diff
@@ -668,26 +735,33 @@ static void hb_sub_status(struct bt_mesh_model *model,
 	param->sub->max = net_buf_simple_pull_u8(buf);
 
 	bt_mesh_msg_ack_ctx_rx(&cli->ack_ctx);
+
+	return 0;
 }
 
 struct hb_pub_param {
 	uint8_t *status;
 	struct bt_mesh_cfg_hb_pub *pub;
 };
 
-static void hb_pub_status(struct bt_mesh_model *model,
-			  struct bt_mesh_msg_ctx *ctx,
-			  struct net_buf_simple *buf)
+static int hb_pub_status(struct bt_mesh_model *model,
+			 struct bt_mesh_msg_ctx *ctx,
+			 struct net_buf_simple *buf)
 {
 	struct hb_pub_param *param;
 
 	BT_DBG("net_idx 0x%04x app_idx 0x%04x src 0x%04x len %u: %s",
 	       ctx->net_idx, ctx->app_idx, ctx->addr, buf->len,
 	       bt_hex(buf->data, buf->len));
 
+	if (buf->len != 10U) {
```

Comment:
Isn't this check redundant since the expected size is part of the `bt_mesh_cfg_cli_op` array?

Reply:
According to spec section 3.7.4.4, models shall ignore messages where "The access message size for the application opcode is incorrect."

The array only checks the minimum length. Perhaps it's worth changing the behavior of the length check to check exact length? We could for instance set it to some magic number like -1 if the length check should be ignored by access, then perform the length check manually for those messages.

Reply:
Yeah, I'd be in favour of that. We could also consider adding a new boolean struct member to indicate whether the length is an exact value or a minimum. Whether we do that or go with `-1` I suppose this still constitutes API breakage which must be documented.